tasks/travis-pr-title-lint.js

```diff
@@ -0,0 +1,159 @@
+var objectValues = require('object.values');
+var objectEntries = require('object.entries');
+
+
+var changelogLint = changelogLintPkg.default;
+var getPreset = changelogLintPkg.getPreset;
+var getConfiguration = changelogLintPkg.getConfiguration;
+var format = changelogLintPkg.format;
+
+
+module.exports = function(grunt) {
+  function countGitMergeCommits() {
+    return new Promise(function(resolve, reject) {
+      exec('git rev-list --count HEAD ^master', function(err, stdout) {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(parseInt(stdout));
+        }
+      });
+    });
+  }
+
+  function getGitLastCommitMessage() {
+    return new Promise(function(resolve, reject) {
+      exec('git show -s --format=%B HEAD', function(err, stdout) {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(stdout);
+        }
+      });
+    });
+  }
+
+  function getPullRequestTitle() {
+    return new Promise(function(resolve, reject) {
+      var pullRequestURLPath = `/repos/${ process.env.TRAVIS_REPO_SLUG
+        }${/pulls/ }${process.env.TRAVIS_PULL_REQUEST }.json`;
+
+      grunt.log.writeln(
+        `Retrieving the pull request title from https://api.github.com${
+          pullRequestURLPath}`
+      );
+
+      https.get({
+        host: 'api.github.com',
+        path: pullRequestURLPath,
+        headers: {
+          'User-Agent': 'mozilla web-ext grunt tasks',
+        },
+      }, function(response) {
+        var body = '';
+        response.on('data', function(data) {
+          body += data;
+        });
+        response.on('error', function(err) {
+          grunt.log.writeln(`Failed during pull request title download: ${
+                            err}`);
+          reject(err);
+        });
+        response.on('end', function() {
+          try {
+            resolve(JSON.parse(body).title);
+          } catch (err) {
+            reject(err);
+          }
+        });
+      });
+    });
+  }
+
+  function lintMessage(message) {
+    return new Promise(function(resolve, reject) {
+      return Promise.all([
+        getPreset('angular'),
+        getConfiguration('.conventional-changelog-lintrc'),
+      ]).then((results) => {
+        var preset = results[0];
+        var configuration = results[1];
+
+        grunt.log.writeln(`Changelog lint input: ${ message}`);
+        return changelogLint(message, {preset, configuration});
+      }).then((report) => {
+        const formatted = format(report, {
+          color: true,
+          signs: [' ', '⚠', '✖'],
+          colors: ['white', 'yellow', 'red'],
+        }).join('\n');
+
+        if (report.valid) {
+          resolve(formatted);
+        } else {
+          reject(formatted);
+        }
+      });
+    });
+  }
+
+  var writeCommitMessagesDocURL =
+        'https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#writing-commit-messages'; // eslint-disable-line max-len
```

Comment:
yeah, this will be definitely in a chapter of my book "when your editor auto-indentation feature works against you" :-D

when I moved all the grunt task code into a separate file, the "auto-indentation on copy & paste" feature decided to mess up my manual indentation :-|